monitoring/mongodb/alerts.yaml

```diff
@@ -139,3 +139,55 @@ groups:
     annotations:
       description: 'MongoDb has low disk space'
       summary: 'MongoDb has low disk space'
+  
+  - alert: MongoDbPodRecovering
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"}[1h]) == 3
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past hour. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time. This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbPodRecovering
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"}[1d]) == 3
+    for: 1d
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past day. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time. This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbInvalidState
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state=~"6|8|10"}[1m]) > 0
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` is repeatedly or continuously in the `{{ $labels.rs_state }}` state. Immediate attention is required. You may collect an sosreport for further analysis."
+      summary: "MongoDB node in an invalid state: 6 (UNKNOWN), 8 (DOWN), 10 (REMOVED)"
+
+
+  - alert: MongoDbPodStartup2
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="5"}[1h]) == 5
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been in the 'STARTUP2' state for an hour. Please ensure that the instance is running properly. You may follow the 'Resync a Data Services MongoDB Member' procedure to unblock this MongoDB instance instance."
+      summary: MongoDB node in STARTUP2 state for too long
+
+  - alert: MongoDbRSNotSynced
+    expr: |
+      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+    for: 10m
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB replica set `{{ $labels.rs_nm }}` is missing `{{ $value }}` SECONDARY members for at least 1 hour. This means that in case of primary loss, there is a risk of metadata loss. Immediate action is required to ensure data integrity."
```

Comment:
Now that we decided to have an alert for the last 10min, the metadata loss is maybe not appropriate...

So just to not stress the user too much (I also updated the duration in the description):

```suggestion
      description: "MongoDB replica set `{{ $labels.rs_nm }}` is missing `{{ $value }}` SECONDARY members for at least 10 minutes. If the member is not recovering, you may collect an sosreport for further analysis."
```

In such case, the severity can be `warning`.

Now, I think it may make sense to add another alert: if the number of secondary is 0 AND the expected number is >= 1 (to not have this alert for single-node platforms), then we are in a situation where we can really have metadataloss if the primary is lost... Something like:

```
  - alert: MongoDbRSNotProtected
    expr: |
      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) == 0
      and (${replicas} > 1)
    for: 10m
    labels:
      severity: critical
    annotations:
      description: "MongoDB replica set `{{ $labels.rs_nm }}` is missing all its SECONDARY members for at least 10 minutes. This means that in case of primary loss, there is a risk of metadata loss. Immediate action is required to ensure data integrity."
      summary: MongoDB replica set is not protected
```

Reply:
https://github.com/scality/Zenko/pull/2142/commits/0ec53245375a1574c5b51e7a08bdaa58d8b395d6#diff-237b96ceb855bab52e3f81d4a49e5e17e07fc7c1102d3dec5b30e461fee6fc69R194-R203